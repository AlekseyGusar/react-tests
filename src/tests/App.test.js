const axios = require("axios");
import DataService from '../dataService';
import renderer from 'react-test-renderer'; 

describe('testing api requests', () => {
  let dataService;

  beforeEach(() => {
    dataService = new DataService(); 
  });

  test('should return gitHub user information', async () => {
    const info = { response: "SomeString" };
    const resp = { data: info };
    jest.spyOn(axios, "get").mockResolvedValueOnce(resp);
    const res = await dataService.getGitHubUser('AlexeyGusar');
    expect(res.data.response).toContain("SomeString");
  });

  test('the fetch fails with an error', async () => {
    jest.spyOn(axios, "get").mockRejectedValueOnce('request error');
    await expect(dataService.getGitHubUser('AlexeyGusar')).rejects.toMatch('request error');
  });
});

describe('snapshot tests', () => {
  test('header renders correctly', () => {
    const wrapper = renderer.create(<h3>GitHub User Info</h3>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  
  test('first item renders correctly', () => {
    const wrapper = renderer.create(<li>"login: AlekseyGusar"</li>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
   
});