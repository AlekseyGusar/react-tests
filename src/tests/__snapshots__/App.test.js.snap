// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests first item renders correctly 1`] = `
<li>
  "login: AlekseyGusar"
</li>
`;

exports[`snapshot tests header renders correctly 1`] = `
<h3>
  GitHub User Info
</h3>
`;
